### PR TITLE
MNT: Remove deprecated blackbody module

### LINKED ISSUE #9434: MNT: Remove deprecated blackbody module and documentation
### Description

Remove deprecated `blackbody` module (deprecated in 4.0) and all references to it in the documentation.

### Additional context

Expanded from what @bsipocz said in https://github.com/astropy/astropy/pull/9282#issuecomment-543966891

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to remove deprecated `astropy.modeling.blackbody` module. It was deprecated in 4.0.

Fixes #9434

### CODE REVIEW

File: docs/modeling/physical_models.rst [hunk 1]
```diff
@@ -77,15 +77,6 @@ give the wavelength and frequency of the maximum for :math:`B_\lambda`
 and :math:`B_\nu`, respectively, calculated using `Wein's Law
```
Comment (dhomeier) [inline docs/modeling/physical_models.rst:77]:
```suggestion
and :math:`B_\nu`, respectively, calculated using `Wien's Law
```
Completely unrelated, just spotted (or rediscovered...) the typo here.

**Comment by dhomeier** [comment]
More of a policy question – this will purge all references to `BlackBody1D` from the docs, right? I suppose the removal of the module proper is on schedule, just wondering if the reference to the old implementation would not still be helpful for people having used this some time ago (I can easily imagine someone updating from 3.x to 4.3 and getting caught by surprise that the old class is gone). I'd even consider leaving a note that `BlackBody1D` was returning normalised bolometric flux as opposed to physical flux density from `BlackBody`. Or is this the changelog entry supposed to be visible enough for backward reference?

File: docs/modeling/physical_models.rst (hunk 1 above)
Comment (pllim) [reply docs/modeling/physical_models.rst]: Ops. Good catch!

File: docs/modeling/physical_models.rst [hunk 2]
```diff
@@ -74,18 +74,9 @@ function gives the bolometric flux using
 The :meth:`~astropy.modeling.physical_models.BlackBody.lambda_max` and
 :meth:`~astropy.modeling.physical_models.BlackBody.nu_max` member functions
 give the wavelength and frequency of the maximum for :math:`B_\lambda`
-and :math:`B_\nu`, respectively, calculated using `Wein's Law
+and :math:`B_\nu`, respectively, calculated using `Wien's Law
 <https://en.wikipedia.org/wiki/Wien%27s_displacement_law>`_.
 
-.. note::
-
-    Prior to v4.0, the ``BlackBody1D`` and the functions ``blackbody_nu`` and ``blackbody_lambda``
```
Comment (pllim) [inline docs/modeling/physical_models.rst:82]: @dhomeier , regarding your comment, would leaving this note in but updating it be sufficient? Or maybe even making this a sub-section and move the `deprecated-blackbody` tag here. What do you think?

Comment (dhomeier) [reply docs/modeling/physical_models.rst:82]: Yes, to clarify after reading up a bit on the discussion in #9434 and #9282, I was just thinking of a brief upgrade guide in that comment. Could perhaps be a sub-section as well, but the comment layout might make it clearer that it is an outdated interface.

Comment (pllim) [reply docs/modeling/physical_models.rst:82]: I added a commit to the doc that points to the deprecated stuff in a previous release. Does this look good to you?

Comment (dhomeier) [reply docs/modeling/physical_models.rst:82]: Ah, forgot that the LTS docs would also still be around! Yes, that's a good solution.

**Review by dhomeier — APPROVED** [review]
Good to go from here; for docs it will still be good to have a second opinion.

**Comment by pllim** [comment]
@karllark did a lot of work on this, so I am sure if anyone has opinion, it is him. 😹

**Review by karllark — APPROVED** [review]
Looks good to me.

File: docs/modeling/physical_models.rst [hunk 3]
```diff
@@ -74,17 +74,18 @@ function gives the bolometric flux using
...
-    were provided.  ``BlackBody1D`` was a more limited blackbody model that was
-    specific to integrated fluxes from sources.  The capabilities of all three
-    can be obtained with :class:`~astropy.modeling.physical_models.BlackBody`.
-    See :doc:`blackbody_deprecated`
-    and `astropy issue #9066 <https://github.com/astropy/astropy/issues/9066>`_ for details.
+Prior to v4.0, blackbody functionality was provided by the
+``astropy.modeling.blackbody`` module, which was then removed in v4.3.
+If you are still using the removed ``blackbody`` module, please see
+`Blackbody Module (deprecated capabilities) <https://docs.astropy.org/en/v4.1/modeling/blackbody_deprecated.html>`_.
```
Comment (bsipocz) [inline docs/modeling/physical_models.rst:88]: Ideally this will need to be cleaned up in 5.0, a new LTS is like a new blank page :)

Comment (pllim) [reply docs/modeling/physical_models.rst:88]: @bsipocz , clean up how? Could you please clarify?

Comment (bsipocz) [reply docs/modeling/physical_models.rst:88]: not mention the old way, and link old docs.

Comment (pllim) [reply docs/modeling/physical_models.rst:88]: Gotcha, thanks! Opened #10986 to track this.

**Pull request merged**